File: app/src/main/java/mobiletest/yucef/ca/mobiletest/model/Resource.java
package mobiletest.yucef.ca.mobiletest.model;

import java.util.List;

import mobiletest.yucef.ca.mobiletest.ResourcesActivity;

public class Resource {
    String id;
    String title;
    String description;
    String photoUrl;
    List<Address> addresses;
    List<ContactInfo> contactInfoList;

    public Resource(String id, String title, String description, String photoUrl, List<Address> addresses, List<ContactInfo> contactInfoList) {
        this.title = title;
        this.id = id;
        this.description = description;
        this.photoUrl = photoUrl;
        this.addresses = addresses;
        this.contactInfoList = contactInfoList;
    }

    public String getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public String getPhotoUrl() {
        return photoUrl;
    }

    public List<Address> getAddresses() {
        return addresses;
    }

    public List<ContactInfo> getContactInfoList() {
        return contactInfoList;
    }
}

File: app/src/main/java/com/quickseries/mobiletest/domain/resources/model/Resource.kt
package com.quickseries.mobiletest.domain.resources.model

import com.quickseries.mobiletest.data.resources.ResourceResponse
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import java.util.*

data class Resource(
    val title: String,
    val description: String,
    val photoUrl: String,
    val lastModified: Date? = null,
    val contactInfo: ContactInfo,
    val addresses: List<Address>,
    val bizHours: BizHours,
    val socialMedia: SocialMedia
)

suspend fun List<ResourceResponse>.toResources() = withContext(Dispatchers.Default) {
    map { resourceResponse ->
        Resource(
            resourceResponse.title.orEmpty(),
            resourceResponse.description.orEmpty(),
            resourceResponse.photoUrl.orEmpty(),
            resourceResponse.updatedAt ?: resourceResponse.createdAt,
            resourceResponse.contactInfo.toContactInfo(),
            resourceResponse.addresses.toAddresses(),
            resourceResponse.bizHours.toBizHours(),
            resourceResponse.socialMedia.toSocialMedia()
        )
    }
}